apps/box/src/components/UsageBar.tsx

```diff
@@ -50,78 +115,112 @@ export const UsageBar = ({ isEditable }: UsageBarProps) => {
     }
   >({
     onStart: (_, ctx) => {
-      ctx.offsetX = translateX.value;
+      ctx.offsetX = dividerX.value;
+      runOnJS(editStartHandler)();
     },
     onActive: (event, ctx) => {
-      translateX.value = clamp(
-        ctx.offsetX + event.translationX,
-        boundsX.value.low,
-        boundsX.value.high
+      divisionPercent.value = clamp(
+        toPercentage(ctx.offsetX + event.translationX, boundsX),
+        clampBoundaries.value.low,
+        clampBoundaries.value.high
       );
     },
+    onEnd: () => {
+      runOnJS(editEndHandler)();
```

Comment:
Is it necessary to run on js thread?

Reply:
Yes, running this on JS so that screens that set up this callback are able to adjust their state, etc.